denso_robot.py

```diff
@@ -65,3 +65,21 @@ def unmount(self, gov_robot, puck_pos: int, pin_pos: int, samp_id: str):
 
     def finish(self):
         ...
+
+    def check_sample_mounted(self, mount, puck_pos, pin_pos):  # is the correct sample present/absent as expected during a mount/unmount?
+        if mount:
+            check_occupied = 1
+        else:
+            check_occupied = 0
+        if int(self.robot.spindle_occupied_sts.get()) == check_occupied and \
+           int(self.robot.puck_num_sel.get()) == puck_pos - 1 and \
```

Comment:
Is this the zero indexed vs. one indexed issue?

Reply:
yes - could add comments in that direction for all of the comparisons